Fix NatSpec json output for "@notice" and "@dev" tags on contract definitions

zone117x opened the pull request:
Fixes issue https://github.com/ethereum/solidity/issues/4446

zone117x committed:
Fix NatSpec json output for "@notice" and "@dev" tags on contract definitions.

ipreferpi reviewed (approved):
👌

axic commented on libsolidity/interface/Natspec.cpp:
```diff
@@ -36,6 +36,10 @@ Json::Value Natspec::userDocumentation(ContractDefinition const& _contractDef)
 	Json::Value doc;
 	Json::Value methods(Json::objectValue);
 
+	auto dev = extractDoc(_contractDef.annotation().docTags, "notice");
```
Why is the variable called `dev`? Shouldn't it be `notice`? Also below in the loop we use an explicit type (`string`).

zone117x replied:
Fixed!

axic commented:
The natspec logic seems to be correct though (compared to the "spec").

zone117x commented:
@axic Thanks for the feedback. Fixed the concerns you brought up. Are we good to merge now?

chriseth commented:
Please add an entry into the changelog, in the "bugfixes" category.

zone117x committed:
Add changelog bugfix

For fixing NatSpec json output for "@notice" and "@dev" tags on contract definitions.

zone117x commented:
@chriseth Done!
